src/assets/woodpeckerCI/woodpeckerCI.md

```diff
@@ -0,0 +1,262 @@
+
+- Enable it.
+
+![Enable Repo](https://raw.githubusercontent.com/JankariTech/blog/woodpecker-ci-starlark/src/assets/woodpeckerCI/images/enable.png)
+
+Woodpecker will automatically add the necessary webhooks to the repo.
+
+
+## Generating the Public Key
+Before running any pipeline, woodpecker server needs send POST request to an external config service which, in our case, is WCCS with all current config file and build information of the current repository. The external service will then send back new pipeline configurations. For example, starlark is being converted into yaml configurations.
+
+Every request sent by Woodpecker is signed using a http-signature by a private key (ed25519) generated on the first start of the Woodpecker server. You can get the public key for the verification of the http-signature from:
+```console
+https://<your-ci-server>/api/signature/public-key
+```
+Store that public key into `keys/public.pem` in the same directory level as your docker compose, it will be needed for signature verification.
+
+## Setting Up WCCS-Woodpecker CI Config Service (Starlark Conversion Service)
+The Woodpecker Config Conversion Service (WCCS) is a lightweight web service created and maintained by [Opencloud-eu](https://opencloud.eu/en). It enables Woodpecker CI to convert pipeline definitions written in Starlark into standard YAML on the fly by receiving a signed POST request from Woodpecker.
+You can easily deploy WCCS using their official Docker image available on Docker Hub: [opencloudeu/wccs](https://hub.docker.com/r/opencloudeu/wccs).
+Update your `docker-compose.yml` to include the WCCS service:
+```yml
+  wccs:
+    image: opencloudeu/wccs:latest
+    container_name: wccs
+    command: server
+    environment:
+      - WCCS_LOG_LEVEL=debug
+      - WCCS_SERVER_PUBLIC_KEY=/keys/public.pem
+    volumes:
+      - /opt/woodpecker/keys:/keys
+      - /etc/ssl/certs:/etc/ssl/certs:ro
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.wccs.rule=Host(`your-wccs-server`)"
+      - "traefik.http.routers.wccs.entrypoints=websecure"
+      - "traefik.http.routers.wccs.tls.certresolver=letsencrypt"
+      - "traefik.http.services.wccs.loadbalancer.server.port=8080"
+    networks:
+      - woodpecker-net
+
+volumes:
+  woodpecker-server-data:
+  woodpecker-agent-config:
+
+networks:
+  woodpecker-net:
+    driver: bridge
+```
+
+## Connecting WCCS with Woodpecker
+To allow Woodpecker to fetch pipeline configs from WCCS, we added this to the server env:
+```console
+WOODPECKER_CONFIG_SERVICE_ENDPOINT=http://wccs:8080/ciconfig
+```
+Woodpecker now sends a signed JSON payload to WCCS whenever a build is triggered. WCCS verifies the signature using the public key and responds with a YAML pipeline based on your .woodpecker.star file.
+
+After conneting WCCS with woodpecker server, there's one more thing that you need to do i.e. to set pipeline config path.
```

Comment:
```suggestion
After conneting WCCS with the woodpecker server, there's one more thing that you need to do i.e. to set pipeline config path, because by default Woodpecker will take .... as its pipeline configurations.
```